kong/db/strategies/postgres/connector.lua

```diff
@@ -371,9 +372,46 @@ function _mt:setkeepalive()
 end
 
 
+function _mt:acquire_lock()
+  if get_phase() == "init" or get_phase() == "init_worker" then
+    return true
+  end
+
+  if not self.sem then
```

Comment:
We should perform this check before the `get_phase()` checks, to avoid unnecessary CPU cycles when `pg_semaphore_max = 0`.